testimony/config.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: UTF-8 -*-
+
+"""
+Parse config for testimony.
+
+Example:
+
+    ---
+    Status:
+        type: choice
+        choices:
+            - Manual
+            - Automated
+    ...
+
+Currently only supported type is 'choice', but in the future more can be added.
+"""
+
+import sys
+import yaml
+
+TOKEN_TYPES = ['choice']
+
+
+def parse_config(filename):
+    """Parse the config."""
+    try:
+        with open(filename, 'r') as fp:
+            data = yaml.load(fp, Loader=yaml.SafeLoader)
+            return {k.lower(): TokenConfig(k, v) for k, v in data.items()}
+    except FileNotFoundError:
+        print('ERROR: File {} not found'.format(filename))
+        sys.exit(1)
+
+
+def update_tokens_dict(tokens_dict, new_keys, default_value=None):
+    """Update dictionary by adding new keys and setting default value."""
+    if isinstance(new_keys, str):
+        new_keys = [token.strip().lower()
+                    for token in new_keys.split(',')
+                    if token.strip()]
+
+    if default_value is None:
+        default_value = {}
+
+    for key in new_keys:
+        if key in tokens_dict:
+            tokens_dict[key].update(default_value)
+        else:
+            tokens_dict[key] = TokenConfig(key, default_value)
+
+
+class TokenConfig(object):
+    """
+    Represent config for one token.
+
+    Curently only checks for value.
+    """
+
+    def __init__(self, name, config):
+        """
+        Initialize token config object.
+
+        Takes name of the token and actual config structure as a param
+        """
+        try:
+            config.get('key')
+        except AttributeError:
+            config = {}
+
+        self.name = name.lower()
+        self.mandatory = config.get('mandatory', False)
+        self.token_type = ''
```

Comment:
Good point. I like `None` better, as it is less limiting for us. If we set default to `'string'`, and then decide that strings should always have some additional option, we would break backwards compatibility. There are no problems like that with `None`.

And once we are sure we don't want any required option for `'string'`, ever, we can just change it.